src/dmd/hdrgen.d

```diff
@@ -250,11 +250,47 @@ public:
     override void visit(WhileStatement s)
     {
         buf.writestring("while (");
-        s.condition.expressionToBuffer(buf, hgs);
+        /**
+         * If we have something like:
+         *
+         * while(true)
+         *     if (auto a = exp) { ... }
+         *
+         * It is possible that this code was lowered
```

Comment:
Foreach statements don't have this problem, do they?

Reply:
Foreach statements have a `Parameter` array that stores all the initialization variables, therefore `hdrgen` can print them directly. Those parameters are needed inside when the `foreach` is lowered to a `for` loop. I can do the same, but that would require storing the parameter in the `WhileStatement` - with the current implementation that would wasteful since generating headers occurs less often than normal compilation.